### ISSUE TITLE
opam-init prepends . to PATH in fish shell

### ISSUE THREAD

**Issue by andrewarchi** [ocaml/opam#4677]
I use the fish shell and noticed that opam-init prepends `.` to PATH. This is a security issue, so I have removed opam-init from my config.fish for the time being. My opam installation is near-fresh as I recently installed it.

Notice that on every prompt, another `.` is added to PATH. I think this is caused by something in [env_hook.fish](https://github.com/ocaml/opam/blob/master/src/state/shellscripts/env_hook.fish).

```sh
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin /Users/USER/.pyenv/shims /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . /Users/USER/.pyenv/shims /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . . /Users/USER/.pyenv/shims /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . . . /Users/USER/.pyenv/shims /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
$
$ echo $fish_user_paths
/Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . . . . . . /Users/USER/.pyenv/shims /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
```

The problematic PATH appears to follow this pattern:

```
. /Users/USER/.opam/4.11.1/bin <repeated .> <other paths added in config.fish> $fish_user_paths $PATH
```

With an empty config.fish, my PATH is just `$fish_user_paths` `$PATH`, as expected:

```sh
$ cat ~/.config/fish/config.fish
$ echo $PATH
/Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
```

If I now clear my config.fish to only contain opam-init, I see the problematic behavior:

```sh
$ cat ~/.config/fish/config.fish 
if test -e ~/.opam/opam-init/init.fish
  source ~/.opam/opam-init/init.fish > /dev/null 2>&1
end
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
$ echo $PATH
. /Users/USER/.opam/4.11.1/bin . . /Users/USER/bin /Users/USER/go/bin /Users/USER/.cargo/bin /usr/local/opt/llvm/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin
```

Here are some system reports:

```sh
$ opam config report
# opam config report
# opam-version      2.0.7 
# self-upgrade      no
# system            arch=x86_64 os=macos os-distribution=homebrew os-version=10.14.6
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              15
# repositories      1 (http) (default repo at e24f43ca)
# pinned            0
# current-switch    4.11.1

$ fish --version
fish, version 3.2.2

$ which opam
/usr/local/bin/opam
```

**Comment by dra27** [comment]
This was fixed in opam 2.0.8

**Comment by andrewarchi** [comment]
Perfect. Now that I upgrade, it works as expected.

**Comment by rjbou** [comment]
fixed in #4084
